src/Microsoft.Diagnostics.Monitoring.EventPipe/Logs/TraceEventExtensions.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Text.Json;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Diagnostics.Monitoring.EventPipe
+{
+    internal static partial class TraceEventExtensions
+    {
+        private static readonly UTF8Encoding s_Utf8Encoding = new(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+        [ThreadStatic]
+        private static LogObject[]? s_ScopeStorage;
```

Comment:
Does this state get cleaned up after each session?

Reply:
Define cleaned up 😄 It gets reused over and over. Never given back. So if you listen to logs for 30 seconds and never again the process will have some memory held it doesn't technically need. It isn't going to be a whole bunch of memory, I don't think. Issue or no?